keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -85,10 +115,13 @@ class MemoryCloudDriverCache(
   /**
    * Convert a suspending function to a [BiFunction] that [AsyncCache.get] expects as its second argument
    *
-   * Uses [Dispatchers.IO] because it assumes that [block] makes I/O calls
+   * Uses a coroutine dispatcher based on the cache's [Executor].
    */
-  private fun <T> asyncify(block: suspend CoroutineScope.() -> T?): BiFunction<String, Executor, CompletableFuture<T?>> =
-    BiFunction { _, _ -> CoroutineScope(Dispatchers.IO).future(block = block) }
+  private fun <K, V> asyncify(block: suspend CoroutineScope.(K) -> V?): BiFunction<K, Executor, CompletableFuture<V?>> =
```

Comment:
Feels like we win Kotlin type signature bingo here.

Reply:
Yeah, I added the key as a generic because initially I was using a different key type for the new cache, but then I looked at how all the other ones were implemented & decided to keep it consistent.